core/src/test/java/org/arquillian/smart/testing/scm/git/GitChangeResolverTest.java

```diff
@@ -59,20 +65,51 @@ public void should_fetch_all_files_from_first_commit_to_given_hash() throws Exce
         this.gitChangeResolver = new GitChangeResolver(gitFolder.getRoot(), "d923b3a", "1ee4abf");
 
         // when
-        final Set<Change> diff = gitChangeResolver.diff();
+        final Set<Change> diff = gitChangeResolver.diff(CUSTOM);
 
         // then
         assertThat(diff).hasSize(18);
     }
 
     @Test
+    public void should_not_applicable_when_git_repository_is_not_initialized() throws IOException {
+        // given
+        gitFolder.delete();
+        gitFolder.create();
+
+        this.gitChangeResolver = new GitChangeResolver(gitFolder.getRoot(), "HEAD", "HEAD~0");
+
+        // when
+        final boolean applicable = gitChangeResolver.isApplicable();
+
+        // then
+        assertThat(applicable).isFalse();
+    }
+
+    @Test
+    public void should_throw_exception_for_fetching_all_changes_when_git_repository_is_not_initialized()
+        throws IOException {
+        // given
+        gitFolder.delete();
+        gitFolder.create();
+
+        this.gitChangeResolver = new GitChangeResolver(gitFolder.getRoot(), "HEAD", "HEAD~0");
+
+        // then
+        thrown.expect(IllegalStateException.class);
+
+        // when
+        gitChangeResolver.diff(CUSTOM);
```

Comment:
Also, verify the case when the strategy name is not specified: 
`gitChangeResolver.diff()` 
In this case, it should return the empty set.